frontend/src/app/search.tsx

```diff
@@ -183,36 +203,33 @@ export default class Search extends Component {
     console.log('handleSelectAll')
     this.setState({check: !this.state.check}, () => {
       this.setState(prevState => {
+        this.state.allPaths=[]
+        this.transientPaths=[]
         const selectAllcheck = this.state.data.map(dataitem => {
               dataitem["checkedItem"] = this.state.check
               console.log(dataitem["pant:transientPath"]+":"+dataitem["checkedItem"])
-              this.state.check?this.state.allPaths.push(dataitem["pant:transientPath"]):delete this.state.allPaths[this.state.allPaths.indexOf(dataitem["pant:transientPath"])]
+              this.state.check?this.state.allPaths.push(dataitem["pant:transientPath"]):this.state.allPaths.splice(this.state.allPaths.indexOf(dataitem["pant:transientPath"]),1)
           return dataitem
         })
         this.transientPaths=this.state.allPaths
-        this.transientPaths.map(e => e === "" ? delete this.transientPaths[this.transientPaths.indexOf(e)] : e)
-        console.log('final transientPaths:'+this.transientPaths)
-        console.log('all paths out:'+this.state.allPaths)
+        this.transientPaths.map(e => e === "" ? this.transientPaths.splice(this.transientPaths.indexOf(e),1) : e)
         if(this.state.check === true){
-          this.setState({countOfCheckedBoxes: this.state.countOfCheckedBoxes+this.state.data.length}, () => {
-                console.log('countOfCheckedBoxes: '+this.state.countOfCheckedBoxes)
+          this.setState({countOfCheckedBoxes: this.state.data.length}, () => {
+            console.log('countOfCheckedBoxes: '+this.state.countOfCheckedBoxes)
                 if(this.state.countOfCheckedBoxes > 0){
                   this.setState({deleteButtonVisible: true})
                 }else{
                   this.setState({deleteButtonVisible: false})
                 }
+                console.log('transientPaths:'+this.transientPaths)
               })
         }else{
-          this.setState({countOfCheckedBoxes: this.state.countOfCheckedBoxes-this.state.data.length}, () => {
-                console.log('countOfCheckedBoxes: '+this.state.countOfCheckedBoxes)
-                if(this.state.countOfCheckedBoxes > 0){
-                  this.setState({deleteButtonVisible: true})
-                }else{
-                  this.setState({deleteButtonVisible: false})
-                }
-              })
-              this.transientPaths = []
-              console.log('transientPaths:'+this.transientPaths)
+          this.setState({countOfCheckedBoxes: 0}, () => {
+            console.log('countOfCheckedBoxes: '+this.state.countOfCheckedBoxes)
+            this.transientPaths = []
+            console.log('transientPaths:'+this.transientPaths)
+            this.setState({deleteButtonVisible: false})
```

Comment:
I remember we had talked about this block. Looks good. :)

Reply:
Yes, this was covered.
Thanks Ben!